### ISSUE TITLE
Gajim carbons are not received in converse.js

### ISSUE THREAD

**Issue by allo-** [conversejs/converse.js#1332]
When I write something in gajim, I cannot see the carbon in converse.js.

I see the carbon in Conversations and I see Conversations carbons in converse.js.
Only gajim to converse.js seem to be broken.

Related bug in the gajim bugtracker: https://dev.gajim.org/gajim/gajim/issues/9447

**Comment by jcbrand** [comment]
Converse can only show the carbon if it actually receives it, so let's check whether that happens.

Set `debug: true` when calling `converse.initialize`, then make sure your console logs verbosely.

Then when sending a message, check in your browser's dev console whether a stanza for that message is received. If so, please paste the stanza here and we can try to figure out why it might be ignored.

**Comment by allo-** [comment]
```
 2018-11-17T19:04:12+01:00 INFO: no requests during idle cycle, sending blank request 
(message sent now)
converse.js:63076  2018-11-17T19:04:17+01:00 INFO: _dataRecv called 
libsignal-protocol.js:36320 New remote ephemeral key
converse.js:63076  2018-11-17T19:04:17+01:00 INFO: no requests during idle cycle, sending blank request 
converse.js:63076  2018-11-17T19:04:17+01:00 INFO: _dataRecv called 
converse.js:63076  2018-11-17T19:04:17+01:00 INFO: no requests during idle cycle, sending blank request 
```

When I enabled ``debug``, I got (before sending or receiving anything) a red message in the chat window:

    Sorry, could not decrypt a received OMEMO message due to an error. MessageCounterError Message key not found. The counter was repeated or the key was not filled.

Probably for some message from MAM. 

The log contains some old messages, which cannot be decrypted and are shown as ``I sent you an OMEMO encrypted message but your client doesn’t seem to support that. Find more information on https://conversations.im/omemo``, but the most recent messages (received, sent, sent from Conversations) are decrypted correctly. The mesages sent from gajim are missing.

**Comment by allo-** [comment]
the libsignal-protocol.js message appears only on the first gajim message, further messages just log the _dataRecv call.

**Comment by jcbrand** [comment]
OK, so this is with OMEMO messages? Are you using the latest version of Converse?

Might be this? #1193

**Comment by allo-** [comment]
I tried the mentioned gajim patch to test, now I cannot open the corresponding conversation in converse.js at all (the chat area stays grey).

In the console I now get the error message:

```
Failed to load resource: the server responded with a status of 404 ()
xmppserver/:1 Uncaught (in promise) DOMException: Failed to load because no supported source was found.
converse.js:63112  2018-11-17T20:11:58+01:00 INFO: _dataRecv called 
converse.js:63107  FATAL: Error: No JID to get vcard for!
    at Object.get (https://xmppserver/chat/converse/dist/converse.js:77441:23)
    at Object.update (https://xmppserver/chat/converse/dist/converse.js:77467:25)
    at child.on.vcard (https://xmppserver/chat/converse/dist/converse.js:77281:55)
    at triggerEvents (https://xmppserver/chat/converse/dist/converse.js:2970:57)
    at triggerApi (https://xmppserver/chat/converse/dist/converse.js:2955:19)
    at eventsApi (https://xmppserver/chat/converse/dist/converse.js:2754:16)
    at child.Events.trigger (https://xmppserver/chat/converse/dist/converse.js:2945:5)
    at child._onModelEvent (https://xmppserver/chat/converse/dist/converse.js:3786:20)
    at triggerEvents (https://xmppserver/chat/converse/dist/converse.js:2971:58)
    at triggerApi (https://xmppserver/chat/converse/dist/converse.js:2956:22)
_converse.log @ converse.js:63107
converse.js:63107  FATAL: error: No JID to get vcard for!
_converse.log @ converse.js:63107
converse.js:63104  2018-11-17T20:11:58+01:00 WARNING: Removing Strophe handlers due to uncaught exception: No JID to get vcard for!


```

**Comment by allo-** [comment]
I didn't had the latest converse.js included. It seems that #1193 fixed it (at least 4.0.5 now shows carbons).

How should the vcard error be fixed? It seems I now will need this workaround as long as the message from the patched gajim is in the archive.

**Comment by allo-** [comment]
Maybe the vCard error is unrelated. I cannot reproduce it reliable when debugging. I removed the patched line and currently it works as well.

I close here now and open another bug if the vcard thing happens more often.
